### PR TITLE
Vir update

### PR DESCRIPTION
**Pull request by manojkrsaini09**
vir update, amc visit update functionality

### CODE REVIEW

File: src/main/java/creativei/controller/AmcContractController.java [hunk 1]
```diff
@@ -209,6 +219,122 @@ ResponseObject updateCustomerAmcContract(@RequestBody String customerAmcStr,
...
+			return ResponseObject.getResponse(ExceptionType.INVALID_METHOD_PARAM.getMessage()
+					, ExceptionType.INVALID_METHOD_PARAM.getCode());
+		} catch (ApplicationLevelException e) {
+			logger.error(e.getMessage(), e);
+			return ResponseObject.getResponse(ExceptionType.GENERAL_ERROR.getMessage()
+					, ExceptionType.GENERAL_ERROR.getCode());
+		}
+	}
+
+	/*@RequestMapping(value = "/vir/save", consumes = "application/json", produces = "application/json" , method = RequestMethod.POST)
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/AmcContractController.java]: Remove commented code.

File: src/main/java/creativei/controller/AmcContractController.java [hunk 2]
```diff
@@ -234,4 +360,45 @@ ResponseObject getAmcDurations(@RequestParam(value = "id", required = true) Stri
 		}
 	}
 
+	@RequestMapping(value = "/customer/amc/visit/vir" , method = RequestMethod.GET)
+	public
+	@ResponseBody
+	ResponseObject getVIRForVisit(@RequestParam(value = "visitId", required = true) String amcVisitIdStr,
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/AmcContractController.java]: Should be a post request as we are saving vir entity here. /customer/amc/visit/vir/generate

File: src/main/java/creativei/dao/AmcVisitDao.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import creativei.entity.AmcVisit;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+@Transactional
+public interface AmcVisitDao {
+    AmcVisit getById(Long id);
+    AmcVisit save(AmcVisit ammcVisit);
+    List<AmcVisit> getByContract(Long amcContarctId);
```
Comment (aayushKhandpur) [inline src/main/java/creativei/dao/AmcVisitDao.java]: getByContractId

File: src/main/resources/public/module/views/amcVisit.tpl.html [hunk 4]
```diff
@@ -0,0 +1,74 @@
+
+<div>
```
Comment (aayushKhandpur) [inline src/main/resources/public/module/views/amcVisit.tpl.html]: Add ui screenshot

File: src/main/java/creativei/controller/AmcContractController.java (hunk 1 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/AmcContractController.java]: done

File: src/main/java/creativei/dao/AmcVisitDao.java (hunk 3 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/dao/AmcVisitDao.java]: done

File: src/main/resources/public/module/views/amcVisit.tpl.html (hunk 4 above)
Comment (manojkrsaini09) [reply src/main/resources/public/module/views/amcVisit.tpl.html]: image

Comment (manojkrsaini09) [reply src/main/resources/public/module/views/amcVisit.tpl.html]: Contract detail page:-
image

Comment (manojkrsaini09) [reply src/main/resources/public/module/views/amcVisit.tpl.html]: Visit Detail Page:-

image

Comment (aayushKhandpur) [reply src/main/resources/public/module/views/amcVisit.tpl.html]: 1. Can we show vir form on the visit create page?
2. Visit table should have grid.
3. Use <h3> tage to show page headings. Refer Product detail page for reference.
4. Dates in the VIR table should be hyperlinks. Show them as hyperlink.
5. Show visit table below Product table.

File: src/main/java/creativei/controller/AmcContractController.java (hunk 2 above)
Comment (aayushKhandpur) [reply src/main/java/creativei/controller/AmcContractController.java]: This is pending.

File: src/main/resources/public/languages/en.json [hunk 5]
```diff
@@ -35,7 +35,9 @@
 		"visit.vir.generate.error" : "Error while Visit info report generation {{errorDetail}}",
 		"visit.vir.update.success" : "Vir updated successfully",
 		"visit.vir.update.error" : "Error in Vir updated {{errorDetail}}",
-		"visit.vir.unsaved" : "Vir is unsaved,Please save vir first."
+		"visit.vir.unsaved" : "Vir is unsaved,Please save vir first.",
```
Comment (aayushKhandpur) [inline src/main/resources/public/languages/en.json]: Use test: "VIR form is not saved."

File: src/main/resources/public/languages/en.json [hunk 6]
```diff
@@ -35,7 +35,9 @@
...
+		"visit.save.daterange.error" : "Visit Date should lies between contract end and start date.",
```
Comment (aayushKhandpur) [inline src/main/resources/public/languages/en.json]: Text: "Visit date not between contract start and end dates."

File: src/main/java/creativei/controller/AmcContractController.java (hunk 2 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/AmcContractController.java]: "customer/amc/visit/vir/generate"
there is no such request mapping now. I think it was unused so I have removed it.

File: src/main/resources/public/module/views/amcVisit.tpl.html (hunk 4 above)
Comment (manojkrsaini09) [reply src/main/resources/public/module/views/amcVisit.tpl.html]: @aayushKhandpur  I have done these changes

**Comment by manojkrsaini09** [comment]
I have done all suggested changes. Please check last commit.

File: src/main/resources/public/module/views/amcVisit.tpl.html (hunk 4 above)
Comment (aayushKhandpur) [reply src/main/resources/public/module/views/amcVisit.tpl.html]: Aproved

**Pull request merged**